service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedSchoolProcessorImpl.java

```diff
@@ -714,6 +717,30 @@ public BookLogListResponse getBookLog(JWTData userData, int schoolId) {
         : new BookLogListResponse(new ArrayList<>());
   }
 
+  @Override
+  public String getReportAsCsv(JWTData userData, int reportId, boolean hasLibrary) {
+    ReportGeneric report;
+    if (hasLibrary) {
+      report =
+          db.selectFrom(SCHOOL_REPORTS_WITH_LIBRARIES)
+              .where(SCHOOL_REPORTS_WITH_LIBRARIES.ID.eq(reportId))
+              .fetchOneInto(ReportWithLibrary.class);
+    } else {
+      report =
+          db.selectFrom(SCHOOL_REPORTS_WITHOUT_LIBRARIES)
+              .where(SCHOOL_REPORTS_WITHOUT_LIBRARIES.ID.eq(reportId))
+              .fetchOneInto(ReportWithoutLibrary.class);
+    }
+    if (report == null) {
+      throw new NoReportByIdFoundException(reportId);
+    }
+    StringBuilder builder = new StringBuilder();
+    builder.append(CsvSerializer.getObjectHeader(report));
+    builder.append(CsvSerializer.toCsv(report));
```

Comment:
Let's wrap this logic in a `try` / `catch`, and throw a custom exception if something goes wrong